lib/dor/text_extraction/abbyy.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Dor
+  module TextExtraction
+    # Build Abbyy tickets for processing
+    class Abbyy
+      attr_reader :filepaths, :druid
+
+      def initialize(filepaths:, druid:)
+        @filepaths = filepaths
+        @druid = druid
+      end
+
+      def write_xml_ticket
+        File.write(abbyy_ticket_filepath, xml_ticket)
+      end
+
+      private
+
+      def abbyy_ticket_filepath
+        File.join(Settings.sdr.abbyy_ticket_path, "#{druid}.xml")
+      end
+
+      def image_abbyy
+        [pdfa_output, alto_output, text_output].join("\n")
+      end
+
+      def pdf_files?
+        File.extname(filepaths.first).strip.downcase[1..] == 'pdf'
```

Comment:
I'm not totally up on the logic here, but do we want to look at all the file paths to see if there is a PDF?

Reply:
I am assuming the filtering of files will happen before we build the ticket and they should all be the same type.

Reply:
That is probably worth a comment, since it is confusing.